quartz/build.ts

```diff
@@ -110,19 +126,148 @@ async function startServing(
     ignoreInitial: true,
   })
 
-  watcher
-    .on("add", (fp) => rebuildFromEntrypoint(fp, "add", clientRefresh, buildData))
-    .on("change", (fp) => rebuildFromEntrypoint(fp, "change", clientRefresh, buildData))
-    .on("unlink", (fp) => rebuildFromEntrypoint(fp, "delete", clientRefresh, buildData))
+  if (argv.fastRebuild) {
+    watcher
+      .on("add", (fp) => partialRebuild(fp, "add", clientRefresh, buildData))
+      .on("change", (fp) => partialRebuild(fp, "change", clientRefresh, buildData))
+      .on("unlink", (fp) => partialRebuild(fp, "delete", clientRefresh, buildData))
+  } else {
+    watcher
+      .on("add", (fp) => rebuildFromEntrypoint(fp, "add", clientRefresh, buildData))
+      .on("change", (fp) => rebuildFromEntrypoint(fp, "change", clientRefresh, buildData))
+      .on("unlink", (fp) => rebuildFromEntrypoint(fp, "delete", clientRefresh, buildData))
+  }
 
   return async () => {
     await watcher.close()
   }
 }
 
+async function partialRebuild(
+  filepath: string,
+  action: FileEvent,
+  clientRefresh: () => void,
+  buildData: BuildData, // note: this function mutates buildData
+) {
+  const { ctx, ignored, depGraphs, contentMap } = buildData
```

Comment:
we also need to acquire and release the build mutex here to prevent overlapping rebuilds from messing each other up

Reply:
also implement the debounce pattern the normal rebuild does

Reply:
Will do. Btw I'm not sure the existing debounce is working. This [line](https://github.com/jackyzha0/quartz/blob/22de92f6c46fd8aae3e803faaea734f28277cfae/quartz/build.ts#L172) doesn't check for 250ms to have elapsed.

Reply:
I read the logic again and don't follow 😅 . Mind explaining how it works?